packages/diagrams/frontend/sirius-components-diagrams/src/renderer/panel/DiagramPanel.tsx

```diff
@@ -116,6 +117,23 @@ export const DiagramPanel = memo(
       applySelection(selection, true);
     }, [selection]);
 
+    const [selectElementsLabel, setSelectElementsLabel] = useState<string | null>('Reveal selected elements');
```

Comment:
I'll extract this part too since it would create a lot of coupling between unrelated parts in the diagram panel.

Reply:
You also don't need a state / useEffect here